virttest/utils_libvirt/libvirt_disk.py

```diff
@@ -538,3 +538,44 @@ def create_remote_disk_by_same_metadata(vm, params):
 
     remote_session.close()
     return blk_source
+
+
+def update_vm_disk_target(vm_name, target_dict, disk_index=0, remove_address=False):
+    """
+    Update vm disk target attributes
+
+    :param vm_name: vm name
+    :param target_dict: dict, new vm disk target attributes
+    :param disk_index: vm disk index to be updated, the index of first disk is 0
+    :param remove_address: True if need to remove device address, otherwise False
+    :return: True if succeeds, otherwise False
+    """
+    vmxml = vm_xml.VMXML.new_from_dumpxml(vm_name)
+
+    try:
+        # Get the disk to be updated
+        devices = vmxml.devices
+        device_index = devices.index(devices.by_device_tag('disk')[disk_index])
+        disk = devices[device_index]
+
+        # Update disk target attributes
+        current_target_dict = disk.target
+        for key, value in target_dict.items():
+            current_target_dict[key] = value
+
+        disk.target = current_target_dict
```

Comment:
```suggestion
        disk.target = dict(disk.target, **target_dict)
```

Reply:
@Yingshun Thanks for your comments and I learned much.

Reply:
@Yingshun I used it in new place, Please help review. Thanks.